index.js

```diff
@@ -123,15 +138,24 @@ async function decodeTorrentFile (torrent) {
   ensure(torrent.info, 'info')
   ensure(torrent.info['name.utf-8'] || torrent.info.name, 'info.name')
   ensure(torrent.info['piece length'], 'info[\'piece length\']')
-  ensure(torrent.info.pieces, 'info.pieces')
 
-  if (torrent.info.files) {
-    torrent.info.files.forEach(file => {
-      ensure(typeof file.length === 'number', 'info.files[0].length')
-      ensure(file['path.utf-8'] || file.path, 'info.files[0].path')
-    })
+  const isV2 = torrent.info['meta version'] === 2
+
+  if (isV2) {
+    // BitTorrent v2 specific validation
+    ensure(torrent.info['file tree'], 'info[\'file tree\']')
   } else {
-    ensure(typeof torrent.info.length === 'number', 'info.length')
+    // BitTorrent v1 validation
```

Comment:
I am reading on phone so not sure if I'm getting the control flow right, but is this v1 validation leg exclusive with v2? We would want to do both in the case of hybrid torrents.